### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by kaadhina**
Porting the customer hot fixes to master.

### CODE REVIEW

**Comment by nigurr** [comment]
LGTM and any UT?

File: Tasks/VsTest/VSTest.ps1 [hunk 1]
```diff
@@ -194,7 +194,7 @@ try
                 $vs15 = Get-VisualStudio_15_0
                 if ($vs15 -and $vs15.Path) {
                     $vstestConsole15 = Get-VSTestConsole15Path -Path $vs15.Path
-                    if ((Test-Leaf -LiteralPath $vstestConsole15)) {
+                    if ((Test-Leaf -LiteralPath $vstestConsole15) -and (InvokeVsTestCmdletHasMember "VSTestLocation")) {
```
Comment (acesiddhu) [inline Tasks/VsTest/VSTest.ps1]: master is v2 version for vstest task. we will only have 109 to do any fixes

**Comment by kaadhina** [comment]
Fixing API version as per @pankagar13

File: Tasks/VsTest/VSTest.ps1 (hunk 1 above)
Comment (kaadhina) [inline Tasks/VsTest/VSTest.ps1]: Already ported for 109

Comment (kaadhina) [inline Tasks/VsTest/VSTest.ps1]: Please file a separate bug for this clean up

**Pull request merged**